src/API/Algorithms/Client.php

```diff
@@ -0,0 +1,419 @@
+
+        return new $class($object, $this);
+    }
+
+    /**
+     * Internal method to wrap items in a collection to resource classes.
+     *
+     * @param  string  $resource  name of the resource class
+     * @param  $object  object collection
+     *
+     * @return array|false|string
+     */
+    private function mapCollection(string $resource, $object)
+    {
+        if ($object === false || is_string($object)) {
+            return $object;
+        }
+
+        $baseResource = __NAMESPACE__ . '\\' . $resource;
+        $resource_class = (class_exists($baseResource)) ? $baseResource : 'Compredict\\API\\Algorithms\\Resources\\' . $resource;
+        $array_of_resources = [];
+
+        foreach ($object as $res) {
+            array_push($array_of_resources, new $resource_class($object, $this));
+        }
+
+        return $array_of_resources;
+    }
+
+    /**
+     * Returns the default collection of algorithms.
+     *
+     * @return mixed array|string list of algorithms.
+     */
+    public function getAlgorithms()
+    {
+        $response = $this->http->GET('/algorithms');
+        if ($response === false || is_string($response)) {
+            return $response;
+        }
+
+        return new Resources\Algorithms($response, $this);
+    }
+
+    /**
+     * Returns the default Resource of algorithm.
+     *
+     * @param String $algorithm_id.
+     * @return Resource/Algorithm object.
+     */
+    public function getAlgorithm($algorithm_id)
+    {
+        $response = $this->http->GET("/algorithms/{$algorithm_id}");
+
+        return $this->mapResource('Algorithm', $response);
+    }
+
+    /**
+     * Returns the default Resource of Task.
+     *
+     * @param String $task_id
+     * @return Resource/Algorithm object.
+     */
+    public function getTaskResult($task_id)
+    {
+        $response = $this->http->GET("/algorithms/tasks/{$task_id}");
+
+        return $this->mapResource('Task', $response);
+    }
+
+    /**
+     * Returns the default Resource of Task after cancelation.
+     *
+     * @param String $task_id
+     * @return Resource/Algorithm object.
+     */
+    public function cancelTask($task_id)
+    {
+        $response = $this->http->DELETE("/algorithms/tasks/{$task_id}");
+
+        return $this->mapResource('Task', $response);
+    }
+
+    /**
+     * Run the algorithm on the given data.
+     *
+     * @param  String  $algorithm_id
+     * @param  Array | String  $data  to predict
+     * @param  Boolean  $evaluate  whether to apply standard evaluation or not.
+     * @param  Boolean  $encrypt  to indicate whether the sent data is encrypt or not.
+     * @param  null  $callback_param  Additional parameters that a requester will receive in the callback url or
+     * when requesting the results.
+     * @param  null  $callback  URL that overrides the main $this->callback for receiving endpoint of data.
+     * @param  string  $file_content_type
```

Comment:
Can you add the docstring for version paramater, it can be either null or of type string and description can be the following:

Specify the algorithm version, if null then the latest version will be requested.